sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -246,9 +246,27 @@ class SessionCatalog(
   def getTableMetadata(name: TableIdentifier): CatalogTable = {
     val db = formatDatabaseName(name.database.getOrElse(getCurrentDatabase))
     val table = formatTableName(name.table)
-    requireDbExists(db)
-    requireTableExists(TableIdentifier(table, Some(db)))
-    externalCatalog.getTable(db, table)
+    val tid = TableIdentifier(table)
+    if (name.database.isEmpty && isTemporaryTable(tid)) {
```

Comment:
Oops. @hvanhovell . I found the original reason.
The logic of `isTemporaryTable` only lookup `current database`. We should support the following. 
```
val m3 = intercept[AnalysisException] {
      catalog.getTableMetadata(TableIdentifier("view1", Some("default")))
    }.getMessage
    assert(m3.contains("Table or view 'view1' not found in database 'default'"))
```